File: example/android/app/src/main/java/com/adyenexample/ForcedAlertModule.java
package com.adyenexample;
import android.app.Activity;
import android.app.AlertDialog;
import android.app.Dialog;
import android.app.FragmentManager;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

import com.facebook.react.bridge.Callback;
import com.facebook.react.bridge.NativeModule;
        import com.facebook.react.bridge.ReactApplicationContext;
        import com.facebook.react.bridge.ReactContext;
        import com.facebook.react.bridge.ReactContextBaseJavaModule;
        import com.facebook.react.bridge.ReactMethod;
        import java.util.Map;
        import java.util.HashMap;

public class ForcedAlertModule extends ReactContextBaseJavaModule {

    public static final String CLOSE = "yourPackageName.CLOSE";

    ForcedAlertModule(ReactApplicationContext context) {
        super(context);
    }

    Intent dialogIntent;

    @NonNull
    @Override
    public String getName() {
        return "ForcedAlertModule";
    }

    protected static Callback onCancel;
    protected static Callback onAgree;

    @ReactMethod
    public void alert(String title, String message, Callback cancelCallBack, Callback okCallBack) {
        onCancel = cancelCallBack;
        onAgree = okCallBack;
        Intent dialogIntent = new Intent(getReactApplicationContext(), AlertDialogActivity.class);
        dialogIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        dialogIntent.putExtra("title", title);
        dialogIntent.putExtra("message", message);
        getCurrentActivity().getApplication().startActivity(dialogIntent);
    }
}

File: example/android/app/src/main/java/com/adyenexample/ForcedAlertModule.kt
package com.adyenexample

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import android.content.Intent
import com.facebook.react.bridge.ReactMethod
import com.adyenexample.ForcedAlertModule
import com.adyenexample.AlertDialogActivity
import com.facebook.react.bridge.Callback

class ForcedAlertModule internal constructor(context: ReactApplicationContext?) :
    ReactContextBaseJavaModule(context) {

    override fun getName(): String {
        return "ForcedAlert"
    }

    @ReactMethod
    fun alert(title: String?, message: String?, cancelCallBack: Callback?, okCallBack: Callback?) {
        onCancel = cancelCallBack
        onAgree = okCallBack
        val dialogIntent = Intent(reactApplicationContext, AlertDialogActivity::class.java)
        dialogIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
        dialogIntent.putExtra("title", title)
        dialogIntent.putExtra("message", message)
        currentActivity!!.application.startActivity(dialogIntent)
    }

    companion object {
        const val CLOSE = "com.adyenexample.CLOSE"
        @JvmField
        var onCancel: Callback? = null
        @JvmField
        var onAgree: Callback? = null
    }
}